src/transform/src/demand.rs

```diff
@@ -32,6 +33,21 @@ use crate::TransformArgs;
 /// not observed in its output. Internal arrangements need not maintain
 /// columns that are no longer required in the join pipeline, which are
 /// those columns not required by the output nor any further equalities.
+///
+/// Nowadays, this transform is mostly obsoleted by `ProjectionPushdown`.
+/// However, one thing that it still does that `ProjectionPushdown` doesn't do
```

Comment:
Do you know about other things that no other transform can do instead of `Demand`?